src/Middleware/HttpTracingMiddleware.php

```diff
@@ -17,40 +18,43 @@
 
 class HttpTracingMiddleware
 {
-    private Tracer $tracer;
     private SpanCreator $spanCreator;
     private TracingRequestGuard $requestGuard;
     private ActiveSpanTraceIdRetriever $activeTraceIdRetriever;
     private TraceIdHeaderRetriever $traceIdHeaderRetriever;
+    private TraceHeaderName $headerNameQuery;
 
     public function __construct(
-        Tracer $tracer,
         SpanCreator $spanCreator,
         TracingRequestGuard $requestGuard,
         ActiveSpanTraceIdRetriever $activeTraceIdRetriever,
-        TraceIdHeaderRetriever $traceIdHeaderRetriever
+        TraceIdHeaderRetriever $traceIdHeaderRetriever,
+        TraceHeaderName $headerName
     ) {
-        $this->tracer = $tracer;
         $this->spanCreator = $spanCreator;
         $this->requestGuard = $requestGuard;
         $this->activeTraceIdRetriever = $activeTraceIdRetriever;
         $this->traceIdHeaderRetriever = $traceIdHeaderRetriever;
+        $this->headerNameQuery = $headerName;
     }
 
+    /** @param Request $request */
     public function handle($request, callable $next)
     {
         if (!$this->requestGuard->allowRequest($request)) {
             return $next($request);
         }
 
-        $traceIdHeader = $this->traceIdHeaderRetriever->retrieve(iterator_to_array($request->headers));
+        $headers = $request->server->all();
+
+        $traceIdHeader = $this->traceIdHeaderRetriever->retrieve($headers, $this->headerNameQuery->name());
```

Comment:
1. Why have you used `server`, not `headers`?

2. `traceIdHeaderRetriever->retrive` is responsible for http-requests and jobs. maybe is it unnecessary? two separated methods will be better, in my mind.

Reply:
1. Replaced to headers. It is easier
2. No, traceIdHeaderRetriever->retrieve() has only one responsibility - fetch header from some carrier. Both casies have carrier with header. But i little bit changed this class.